src/_includes/components/events-banner.njk

```diff
@@ -1,15 +1,63 @@
+{% set colors = ["bg-indigo-700 hover:bg-indigo-600", "bg-indigo-800 hover:bg-indigo-700", "bg-indigo-900 hover:bg-indigo-800"] %}
+
+<div class="event-banner-container bg-indigo-700 hidden sm:block">
+    {% for event in events %}
+    {% if event.link %}
+        {% set color = colors[loop.index0 % colors.length] %}
+        <a href="{{ event.link }}" class="event-banner hidden sm:flex text-center w-full py-1 pb-3 sm:py-3 {{ color }} text-indigo-100 text-sm px-2 gap-1 sm:gap-2 flex-col sm:flex-row items-center justify-center">
+            <span class="hidden sm:block font-semibold">{{ event.type }}</span>
+            <span class="hidden sm:block">-</span>
+            <span class="">{{ event.title }}</span>
+            <span class="hidden sm:block">-</span>
+            <span class="flex items-center gap-1 font-medium underline" >
+                {{ event.buttonText }} {% include "components/icons/chevron-right-sm.svg" %}
+            </span>
+        </a>
+        {% endif %}
+    {% endfor %}
+</div>
+
 <script>
-</script>
+    window.onload = function() {
+        const banners = document.querySelectorAll('.event-banner');
+        const container = document.querySelector('.event-banner-container');
+        let currentBanner = 0;
+
+        // Change the container color to match the current banner
+        function changeContainerColor(banner) {
+            const colorClass = banner.className.match(/bg-\w+-\d+/)[0];
+            const hoverColorClass = banner.className.match(/hover:bg-\w+-\d+/)[0];
+            container.className = 'event-banner-container ' + colorClass + ' ' + hoverColorClass + ' hidden sm:block';
+        }
+
+        // All banners are hidden by default
+        banners.forEach(banner => banner.style.visibility = 'hidden');
 
-<a data-action="event-banner" href="/webinars/2024/deploy-flowfuse-on-industrial-iot-with-ncd-io/"
-    class="hidden md:flex text-center w-full py-1 pb-3 sm:py-3 bg-indigo-700 text-indigo-100 text-sm px-2 gap-1 sm:gap-2 flex-col sm:flex-row items-center justify-center hover:bg-indigo-600">
-    <span class="hidden sm:block font-semibold">Webinar</span>
-    <span class="hidden sm:block">-</span>
-    <span class="">     
-        Deploy FlowFuse on Industrial IoT with NCD.io
-    </span>
-    <span class="hidden sm:block">-</span>
-    <span class="flex items-center gap-1 font-medium underline" >
-        Learn more {% include "components/icons/chevron-right-sm.svg" %}
-    </span>
-</a>
+        // Show the first banner
+        banners[0].style.visibility = 'visible';
+        changeContainerColor(banners[0]);
+
+        // If there is more than one banner, change the visible banner every 5 seconds
+        if (banners.length > 1) {
```

Comment:
Do we need this clause? A transition from one to the same other should be fine?

Reply:
I think the transition is meant to grab the user's attention, conveying something like _the event you saw in the banner when the page first loaded has changed somehow, take a look, there's something different_  and that's why I added this. I don't think we should be grabbing their attention again when there's nothing new to say.